simple_parsing/wrappers/field_wrapper.py

```diff
@@ -164,61 +169,27 @@ def get_arg_options(self) -> Dict[str, Any]:
             # automatically adds the (default: '123')
             _arg_options["help"] = " "
 
-        if utils.is_choice(self.field):
-            _arg_options["type"] = str
-            _arg_options["choices"] = list(self.choices)
+        if self.is_choice:
+            if isinstance(self.choices, dict):
+                _arg_options["type"] = str
+                _arg_options["choices"] = list(self.choices.values())
+            else:
+                assert len(set(type(choice) for choice in self.choices)) == 1
```

Comment:
Here you're asserting that all the options are of the same type, right? I agree, fields like `Literal["cat", 1.0, True]` probably shouldn't be supported. However, this should raise some type of error with a message explaining this.

Reply:
Technically the PEP allows mixing different types in the Literal declaration, but I couldn't find a way to express this to argparse. Maybe you know how to do this.

Reply:
Nono it's fine for Simple-Parsing to choose not to support these kinds of fields, even though it's possible for them to exist.
This just needs to raise an error with a good message so that the users know that's not supported.